pinot-segment-local/src/main/java/org/apache/pinot/segment/local/realtime/impl/forward/FixedByteSVMutableForwardIndex.java

```diff
@@ -81,7 +81,10 @@ public FixedByteSVMutableForwardIndex(boolean dictionaryEncoded, DataType stored
       _valueSizeInBytes = storedType.size();
     }
     _numRowsPerChunk = numRowsPerChunk;
-    _chunkSizeInBytes = numRowsPerChunk * _valueSizeInBytes;
+    _chunkSizeInBytes = (long) numRowsPerChunk * _valueSizeInBytes;
+    Preconditions.checkState(_chunkSizeInBytes < Integer.MAX_VALUE,
```

Comment:
Instead of a precondition here, we should perhaps understand what table configuration led to this and restrict this at the source. What do you think?

Reply:
That makes sense. Removed the precondition.